### ISSUE TITLE
Runner image customization

### ISSUE THREAD

**Issue by kichik** [CloudSnorkel/cdk-github-runners#26]
User should be able to install additional dependencies, setup internal repositories, and in general fully customize the image to their needs.

**Comment by sercantor** [comment]
@kichik does this include build flags? my actions job fails, and when I look at the logs on CodeBuild, there's this error on provisioning stage:
```
SINGLE_BUILD_CONTAINER_DEAD: Build container found dead before completing the build. Build container died because it was out of memory, or the Docker image is not supported
```
I assume this is because I'm using an M1 processor, would be really cool to add something like `DockerBuildAssetOptions` so we could choose the build platform

**Comment by kichik** [comment]
@sercantor yes it will. I should have something you can check out in a few days. It's almost ready. 

But for now, maybe it can be a simple fix like https://github.com/CloudSnorkel/cdk-github-runners/pull/22 for CodeBuild. Can you try a similar change?

**Comment by kichik** [comment]
@sercantor can you please take a look at #47? It should allow customization. It also builds the image in CodeBuild instead of locally. That means you won't need to use a different computer and cross-platform builds should be much easier.

**Comment by sercantor** [comment]
that is great progress! it's super useful to have all these options for a more granular approach, many thanks. 🙇‍♂️
